api/v1alpha2/conversion_scope.go

```diff
@@ -0,0 +1,35 @@
+package v1alpha2
+
+import (
+	"reflect"
+
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+)
+
+//Empty implementation of conversion scope to avoid passing a nil argument to the autogenerated conversion functions
```

Comment:
I believe we always pass `nil` so this likely is not necessary

Reply:
It is being explicitly used in some of the conversions ([example](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/83df3c2746f7eee7675b6ebedd77b4cb85837b4a/api/v1alpha2/zz_generated.conversion.go#L482)) and passing a nil scope results in a nil pointer panic. 
Would love to consider any other suggestions you may have!

Reply:
Weird, that type of code (calling `s.Convert()`) isn't done anywhere else in any of our generated conversion functions. I feel like we're missing some code gen annotations somewhere. Need to do some digging.

Reply:
@detiber @vincepri this is a potential issue. We can't have a no-op scope because it's being used in the generated conversion functions. I'm not currently sure how to either create a scope (it's private in the scheme conversion code) or fix the generation to avoid this. I've asked in Slack [here](https://kubernetes.slack.com/archives/C0EG7JC6T/p1582140633374300) and am awaiting replies. Wanted you to be aware.